main/eval/src/mill/eval/Evaluator.scala

```diff
@@ -66,6 +66,7 @@ object Evaluator {
   // Until we migrate our CLI parsing off of Scopt (so we can pass the BaseModule
   // in directly) we are forced to pass it in via a ThreadLocal
   val currentEvaluator = new DynamicVariable[mill.eval.Evaluator](null)
+  val allEvaluators = new DynamicVariable[Seq[mill.eval.Evaluator]](null)
```

Comment:
Can we base `currentEvalutor` on the value of `allEvaluators` and make it a `def`?

Reply:
We can. Not sure if we should; on one hand making it a def would be less redundant state, on the other hand the current setup is more direct. TBH since it doesn't make a difference now either way, I'm inclined to leave it unchanged for the time being

Reply:
That's fine, too.